src/modules/navigator/takeoff.cpp

```diff
@@ -96,62 +96,48 @@ Takeoff::on_active()
 void
 Takeoff::set_takeoff_position()
 {
-	struct position_setpoint_triplet_s *rep = _navigator->get_takeoff_triplet();
-
-	float abs_altitude = 0.0f;
-
-	float min_abs_altitude;
-
-	// TODO: review this, comments are talking about home pos, the validity is checked but the
-	// current altitude is used instead. Also, the "else" case does not consider the current altitude at all.
-	if (_navigator->home_alt_valid()) { //only use home position if it is valid
-		min_abs_altitude = _navigator->get_global_position()->alt + _navigator->get_takeoff_min_alt();
+	if (!_navigator->home_alt_valid()) {
+		mavlink_log_info(_navigator->get_mavlink_log_pub(),
+				 "Home altitude not valid, abort takeoff\t");
 
-	} else { //e.g. flow
-		min_abs_altitude = _navigator->get_takeoff_min_alt();
+		events::send(events::ID("navigator_takeoff_abort_no_valid_home_alt"), {events::LogLevel::Critical, events::LogInternal::Warning},
+			     "Home altitude not valid, abort takeoff");
+		return;
 	}
 
-	// Use altitude if it has been set. If home position is invalid use min_abs_altitude
-	events::LogLevel log_level = events::LogLevel::Disabled;
+	struct position_setpoint_triplet_s *rep = _navigator->get_takeoff_triplet();
 
-	if (rep->current.valid && PX4_ISFINITE(rep->current.alt) && _navigator->home_alt_valid()) {
-		abs_altitude = rep->current.alt;
+	float takeoff_altitude_amsl = 0.f;
 
-		// If the altitude suggestion is lower than home + minimum clearance, raise it and complain.
-		if (abs_altitude < min_abs_altitude) {
-			if (abs_altitude < min_abs_altitude - 0.1f) { // don't complain if difference is smaller than 10cm
-				mavlink_log_critical(_navigator->get_mavlink_log_pub(),
-						     "Using minimum takeoff altitude: %.2f m\t", (double)_navigator->get_takeoff_min_alt());
-				log_level = events::LogLevel::Warning;
-			}
+	if (rep->current.valid && PX4_ISFINITE(rep->current.alt)) {
+		takeoff_altitude_amsl = rep->current.alt;
 
-			abs_altitude = min_abs_altitude;
+		// If the altitude suggestion is lower than home, notify and abort
+		if (takeoff_altitude_amsl < _navigator->get_home_position()->alt) {
+			mavlink_log_critical(_navigator->get_mavlink_log_pub(),
+					     "abort takeoff \t");
```

Comment:
No event for this error. And does it "abort" if the `takeoff_altitude_amsl` stays low?

Reply:
Removed in 3d45fedf5c399595b885b522abe740dfe9436732